memory/test/test_memory.py

```diff
@@ -33,28 +33,40 @@ def setUp(self):
         token = uuid.uuid4()
         self.test_path = f'gs://{bucket_name}/memory-tests/{token}'
 
-        self.fs = GCS(concurrent.futures.ThreadPoolExecutor(), project=os.environ['PROJECT'])
+        self.fs = GoogleStorageAsyncFS(project=os.environ['PROJECT'])
+        self.sem = asyncio.Semaphore(50)
         self.client = BlockingMemoryClient(fs=self.fs)
         self.temp_files = set()
 
     def tearDown(self):
-        async_to_blocking(self.fs.delete_gs_files(self.test_path))
+        async def delete_files(url):
+            async with self.sem:
+                await self.fs.rmtree(self.sem, url)
+
+        async_to_blocking(delete_files(self.test_path))
         self.client.close()
 
-    def add_temp_file_from_string(self, name: str, str_value: str):
+    async def add_temp_file_from_string(self, name: str, str_value: bytes):
         handle = f'{self.test_path}/{name}'
-        self.fs._write_gs_file_from_string(handle, str_value)
+
+        async with await self.fs.create(handle) as f:
+            await f.write(str_value)
+
         return handle
 
     def test_non_existent(self):
         for _ in range(3):
             self.assertIsNone(self.client._get_file_if_exists(f'{self.test_path}/nonexistent'))
 
     def test_small_write_around(self):
```

Comment:
Heh. That's super frustrating. As far as I can tell, unittest doesn't provide great support for async tests. Pytest works fine:
```
pytestmark = pytest.mark.asyncio  # once per file

async def test():
    assert 3 == 4
```

In the future, when we need to add new async test suites we should use pytest, replacing class fields with pytest fixtures. That'll ensure we can easily test async code without needing to make these blocking shim wrappers.